scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -799,8 +804,271 @@ cncliPTsendslots() {
 
 #################################
 
+cncliEpochData() {
+
+  proc_msg="~ CNCLI epochdata table load started ~"
+
+  getNodeMetrics
+
+  if ! cncliDBinSync; then
+    echo ${proc_msg}
+    echo "CNCLI DB out of sync :( [$(printf "%2.4f %%" ${cncli_sync_prog})] ... check cncli sync service!"
+    exit 1
+  else
+    echo ${proc_msg}
+    echo
+    checkEpochSlot "${choose_func}"
+  fi
+
+}
+
+loadEpochData() {
+  [[ -z "${POOL_VRF_SKEY}" ]] && POOL_VRF_SKEY="${POOL_DIR}/${POOL_VRF_SK_FILENAME}"
+  [[ -z "${CNCLI_DIR}" ]] && CNCLI_DIR="${CNODE_HOME}/guild-db/cncli"
+  [[ -z "${BLOCKLOG_DIR}" ]] && BLOCKLOG_DB="${CNODE_HOME}/guild-db/blocklog"
+
+  CNCLI_DB="${CNCLI_DIR}/cncli.db"
+  BLOCKLOG_DB="${BLOCKLOG_DIR}/blocklog.db"
+
+  cncliParams="--db ${CNCLI_DB} --byron-genesis ${BYRON_GENESIS_JSON} --shelley-genesis ${GENESIS_JSON} --pool-id ${POOL_ID} --pool-vrf-skey ${POOL_VRF_SKEY} --tz UTC"
+  csvdir=/tmp ; tmpcsv="${csvdir}/epochdata_tmp.csv" ; csvfile="${csvdir}/epochdata.csv" ; onerow_csv="${csvdir}/one_epochdata.csv"
+  > "$tmpcsv" ; > "$csvfile" ; > "$onerow_csv"
+
+allEpochsInBlocklog() {
+  EPOCHS=$(sqlite3 "$BLOCKLOG_DB" "SELECT replace(group_concat(DISTINCT epoch ORDER BY epoch ASC), ',', ' ') AS EPOCHS FROM blocklog;")
+}
+
+getKoiosDataHist() {
+  [[ -z ${KOIOS_API} ]] && return 1
+
+  if ! pool_hist=$(curl -sSL -f "${KOIOS_API}/pool_history?_pool_bech32=${POOL_ID_BECH32}&_epoch_no=${EPOCH}" 2>&1); then
+    echo "ERROR: Koios pool_stake_snapshot history query failed."
+    return 1
+  fi
+
+  if ! epoch_hist=$(curl -sSL -f "${KOIOS_API}/epoch_info?_epoch_no=${EPOCH}" 2>&1); then
+    echo "ERROR: Koios epoch_stake_snapshot history query failed."
+    return 1
+  fi
+
+  pool_stake_hist=$(jq -r '.[].active_stake' <<< "${pool_hist}")
+  active_stake_hist=$(jq -r '.[].active_stake' <<< "${epoch_hist}")
+
+  return 0
+}
+
+checkEpochSlot() {
+  getNodeMetrics
+  local func=$1
+  local epochSlot=${slot_in_epoch}
+
+  if [[ $epochSlot -ge 0 && $epochSlot -le 21600 ]]; then
```

Comment:
hardcoded mainnet value, lets make this dynamic based on values from genesis to be compatible with other networks. `read_genesis` function in env file populates various variables with data that could be used for this.

Reply:
Thank you, in the following days I'll review the above requirements
cheers,
Manuel